src/core/components/content-type.jsx

```diff
@@ -24,38 +23,51 @@ export default class ContentType extends React.Component {
   }
 
   componentDidMount() {
-    // Needed to populate the form, initially
-    if(this.props.contentTypes) {
-      this.props.onChange(this.props.contentTypes.first())
+    // Populate the form initially
+    const { contentTypes, onChange } = this.props
+    if (contentTypes && contentTypes.size) {
+      onChange(contentTypes.first())
     }
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    if(!nextProps.contentTypes || !nextProps.contentTypes.size) {
+  componentDidUpdate(prevProps) {
+    const { contentTypes, value, onChange } = this.props
+
+    if (!contentTypes || !contentTypes.size) {
       return
     }
 
-    if(!nextProps.contentTypes.includes(nextProps.value)) {
-      nextProps.onChange(nextProps.contentTypes.first())
+    if (contentTypes !== prevProps.contentTypes || !contentTypes.includes(value)) {
```

Comment:
The additional condition seems to be causing infinite re-renders, at least when testing it locally, as `contentTypes` is not of a primitive type.

```suggestion
    if (!contentTypes.includes(value)) {
```

Reply:
Thank you, I've made the changes and think I've removed everything unrelated to the issue.